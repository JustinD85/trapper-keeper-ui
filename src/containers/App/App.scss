.App {
  background: url('../images/pencils-background.png');
  background-attachment: fixed;
  background-repeat: no-repeat;
  background-size: cover;
  min-height: 1000px;
}
