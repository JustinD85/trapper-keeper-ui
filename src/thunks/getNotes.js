import  { addNotes }  from '../actions';
import { fetchCall } from '../utils/api';

export const getNotes = (url) => {
  return async (dispatch) => {
    const data = await fetchCall(url)
    console.log('from thunk', data)
    dispatch(addNotes(data))   
  }
}