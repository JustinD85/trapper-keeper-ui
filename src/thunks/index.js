import {
  addAllNotes,
  setLoading,
  setError,
  addNewNote,
  updateNote,
  removeNote
} from "../actions";
import { fetchCall } from "../utils/api";

const baseUrl = "http://localhost:3001/notes";

export const getOptions = (method, data) => ({
  method,
  headers: {
    "Content-Type": "application/json"
  },
  body: JSON.stringify(data)
});

export const getNotes = () => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const notes = await fetchCall(baseUrl);
      dispatch(setLoading(false));
      dispatch(addAllNotes(notes));
    } catch (error) {
      dispatch(setError("Error fetching notes"));
    }
  };
};

export const postNote = newNote => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      let note = await fetchCall(baseUrl, getOptions("POST", newNote));
      dispatch(setLoading(false));
      dispatch(addNewNote(note));
    } catch (error) {
      dispatch(setError(error.message));
    }
  };
};

export const putNote = updatedNote => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const note = await fetchCall(
        `${baseUrl}/${updatedNote.id}`,
        getOptions("PUT", updatedNote)
      );
      dispatch(setLoading(false));
      dispatch(updateNote(note));
    } catch (error) {
      dispatch(setError(error.message));
    }
  };
};

export const deleteNote = newNote => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      await fetchCall(`${baseUrl}/${newNote.id}`, { method: 'DELETE' });
      dispatch(setLoading(false));
      dispatch(removeNote(newNote.id));
    } catch (error) {
      dispatch(setError(error.message));
    }
  };
};
