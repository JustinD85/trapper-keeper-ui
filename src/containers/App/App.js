import React, { Component } from "react";
import { NoteArea, CreateNote } from "../";
import { connect } from "react-redux";
import { getNotes } from "../../thunks/";

class App extends Component {
  constructor() {
    super();
    this.state = {
      showPopup: false
    };
  }
  componentDidMount = () => {
    this.fetchNotesOnMount();
  };

  fetchNotesOnMount = () => {
    try {
      this.props.getNotes("http://localhost:3001/notes");
    } catch (error) {
      console.log("you have an error", error);
    }
  };

  handleClick = () => this.setState({ showPopup: !this.state.showPopup });
  render() {
    const { showPopup } = this.state;
    return (
      <div className="App">
        <h1 className="title">Trapper-Keeper</h1>
        <div onClick={this.handleClick}>Add Note</div>
        <CreateNote canRender={showPopup} />
        <NoteArea />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  getNotes: url => dispatch(getNotes(url)),
});

export default connect(
  null,
  mapDispatchToProps
)(App);
