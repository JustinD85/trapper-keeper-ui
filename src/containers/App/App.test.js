import React from "react";
import { shallow } from "enzyme";
import { App, mapDispatchToProps } from "./App";
import { getNotes } from "../../thunks";

jest.mock("../../thunks");
getNotes.mockReturnValue({
  title: "Hello",
  noteItems: [{ text: "world", id: 5, isCompleted: "true" }],
  id: 1
});

const getNotesMock = jest.fn();

describe("App", () => {
  let wrapper;
  window.fetch = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<App getNotes={getNotesMock} />);
  });

  it("should match snapshot and not error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call getNotes on componentDidMount", () => {
    expect(wrapper.instance().props.getNotes).toHaveBeenCalled();
  });

  describe("mapDispatchToProps", () => {
    let mockDispatch = jest.fn();
    let props = mapDispatchToProps(mockDispatch);

    it("should call dispatch with correct params", () => {
      //setup
      const expected = {
        title: "Hello",
        noteItems: [{ text: "world", id: 5, isCompleted: "true" }],
        id: 1
      };
      
      //execution
      props.getNotes();

      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });
  });
});
