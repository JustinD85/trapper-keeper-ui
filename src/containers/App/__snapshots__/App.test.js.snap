// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot and not error 1`] = `
<div
  className="App"
>
  <Route
    component={[Function]}
    path="/"
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/new-note"
    />
    <Route
      path="/notes/:id"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</div>
`;
